src/screens/ExchangeModal.tsx

```diff
@@ -642,13 +647,25 @@ export default function ExchangeModal({
         ? NativeModules.NotificationManager
         : null;
       try {
-        const wallet = await loadWallet();
+        let wallet = await loadWallet();
         if (!wallet) {
           setIsAuthorizing(false);
           logger.sentry(`aborting ${type} due to missing wallet`);
           return false;
         }
 
+        // Switch to the flashbots provider if enabled
+        // TODO(skylarbarrera): need to check if ledger and handle differently here
+        if (
+          flashbots &&
+          currentNetwork === Network.mainnet &&
+          wallet instanceof Wallet
+        ) {
+          logger.debug('flashbots provider being set on mainnet');
+          const flashbotsProvider = await getFlashbotsProvider();
+          wallet = new Wallet((wallet as Wallet).privateKey, flashbotsProvider);
+        }
```

Comment:
Nice. So if `wallet instanceof Wallet`, then we don't need the type cast `wallet as Wallet` right? Or do we need the opposite, in it's a `Signer`, then force it into a `Wallet`.